lib/puppet/configurer/plugin_handler.rb

```diff
@@ -11,13 +11,13 @@ def initialize(factory)
   # Retrieve facts from the central server.
   def download_plugins(environment)
     plugin_downloader = @factory.create_plugin_downloader(environment)
```

Comment:
The factory pattern here seems so unnecessary. 🤷‍♂️

Reply:
I agree... I think I might actually rip it out, in that case. We gotta start somewhere cleaning this stuff up.

Reply:
I did this refactor here: https://github.com/Magisus/puppet/commit/902316a5d173864e6ff17b6b4251ac12b08560fe @branan was worried enough about changing the API that I think merging this in a Z release might be a bad idea, but it'd be nice to merge this into master once this PR has been merged up.